src/map/script.c

```diff
@@ -17084,6 +17084,1021 @@ BUILDIN(getunittype) {
+		case UDT_INT:         nd->status.int_ = (unsigned short) value; status->calc_misc(bl, &nd->status, nd->level); break;
+		case UDT_DEX:         nd->status.dex = (unsigned short) value; status->calc_misc(bl, &nd->status, nd->level); break;
+		case UDT_LUK:         nd->status.luk = (unsigned short) value; status->calc_misc(bl, &nd->status, nd->level); break;
+		case UDT_STATADD:     nd->stat_point = (unsigned int) value; break;
+		case UDT_ATKRANGE:    nd->status.rhw.range = (unsigned short) value; break;
+		case UDT_ATKMIN:      nd->status.rhw.atk = (unsigned short) value; break;
+		case UDT_ATKMAX:      nd->status.rhw.atk2 = (unsigned short) value; break;
+		case UDT_MATKMIN:     nd->status.matk_min = (unsigned short) value; break;
+		case UDT_MATKMAX:     nd->status.matk_max = (unsigned short) value; break;
+		case UDT_DEF:         nd->status.def = (defType) value; break;
+		case UDT_MDEF:        nd->status.mdef = (defType) value; break;
+		case UDT_HIT:         nd->status.hit = (short) value; break;
+		case UDT_FLEE:        nd->status.flee = (short) value; break;
+		case UDT_PDODGE:      nd->status.flee2 = (short) value; break;
+		case UDT_CRIT:        nd->status.cri = (short) value; break;
+		case UDT_RACE:        nd->status.race = (unsigned char) value; break;
+		case UDT_ELETYPE:     nd->status.def_ele = (unsigned char) value; break;
+		case UDT_ELELEVEL:    nd->status.ele_lv = (unsigned char) value; break;
+		case UDT_AMOTION:     nd->status.amotion = (short) value; break;
+		case UDT_ADELAY:      nd->status.adelay = (short) value; break;
+		case UDT_DMOTION:     nd->status.dmotion = (short) value; break;
+		default:
+			ShowWarning("buildin_setunitdata: Invalid data type '%s' for NPC unit.\n", udtype);
+			script_pushint(st, 0);
+			return false;
+		}
+	}
+		break;
+	default:
+		ShowError("buildin_setunitdata: Unknown object!\n");
+		script_pushint(st, 0);
+		return false;
+	} // end of bl->type switch
+	
+	script_pushint(st, 1);
+	return true;
+}
+
+/**
+ * Retrieves real-time data for a game object.
+ * Getunitdata <GUID>,<DataType>{,<Variable>}
+ * @param1  GUID        Game object unique Id.
+ * @param2  DataType    Type of Data to be set for the unit.
+ * @param3  Variable    array reference to store data into. (used for UDT_MAPIDXY)
+ * @return  0 on failure, <value> on success
+ */
+BUILDIN(getunitdata)
+{
+	struct block_list *bl;
+	const char *udtype = NULL;
+	struct map_session_data *sd = NULL;
+	int type = 0;
+	char* name = NULL;
+	struct script_data *data = script_hasdata(st,4)?script_getdata(st, 4):NULL;
+	
+	bl = map->id2bl(script_getnum(st, 2));
+	
+	if (bl == NULL) {
+		ShowWarning("buildin_getunitdata: Error in finding object with given GID %d!\n", script_getnum(st, 2));
+		script_pushint(st, 0);
+		return false;
+	} else if (!script_isstring(st, 3) || (udtype = script_getstr(st, 3)) == NULL || !script->get_constant(udtype, &type)) {
+		ShowWarning("buildin_getunitdata: Undefined Unit Data Type %s set.\n", udtype);
```

Comment:
`udtype` can still be `NULL` here, you're assigning to it as part of a conditional dependent expression in the line above